packages/backend-api/src/Controller/V1/Product/ProductApiDataValidator.php

```diff
@@ -0,0 +1,172 @@
+                    ])
+                ),
+                'sellingFrom' => new Constraints\Optional(
+                    new Constraints\DateTime([
+                        'format' => DateTime::ATOM,
+                        'message' => 'The value {{ value }} is not a valid DateTime::ATOM format.',
+                    ])
+                ),
+                'sellingTo' => new Constraints\Optional(
+                    new Constraints\DateTime([
+                        'format' => DateTime::ATOM,
+                        'message' => 'The value {{ value }} is not a valid DateTime::ATOM format.',
+                    ])
+                ),
+                'catnum' => new Constraints\Optional(
+                    new Constraints\Length([
+                        'max' => 100,
+                        'maxMessage' => 'The value {{ value }} cannot be longer then {{ limit }} characters',
+                    ])
+                ),
+                'ean' => new Constraints\Optional(
+                    new Constraints\Length([
+                        'max' => 100,
+                        'maxMessage' => 'The value {{ value }} cannot be longer then {{ limit }} characters',
+                    ])
+                ),
+                'partno' => new Constraints\Optional(
+                    new Constraints\Length([
+                        'max' => 100,
+                        'maxMessage' => 'The value {{ value }} cannot be longer then {{ limit }} characters',
+                    ])
+                ),
+                'shortDescription' => new Constraints\Optional(
+                    new Constraints\Collection([
+                        'fields' => $descriptionFields,
+                        'allowMissingFields' => true,
+                        'allowExtraFields' => false,
+                    ])
+                ),
+                'longDescription' => new Constraints\Optional(
+                    new Constraints\Collection([
+                        'fields' => $descriptionFields,
+                        'allowMissingFields' => true,
+                        'allowExtraFields' => false,
+                    ])
+                ),
+            ],
+            'allowExtraFields' => false,
+        ];
+    }
+
+    /**
+     * @return array
+     */
+    protected function createNameConstraintDefinition(): array
+    {
+        $nameFields = [];
+        foreach ($this->domain->getAllLocales() as $locale) {
+            $nameFields[$locale] = new Constraints\Length([
+                'max' => 255,
+                'maxMessage' => 'The value {{ value }} cannot be longer then {{ limit }} characters',
+            ]);
+        }
+
+        return $nameFields;
+    }
+
+    /**
+     * @return array
+     */
+    protected function createDescriptionConstraintDefinition(): array
+    {
+        $descriptionFields = [];
+        foreach ($this->domain->getAllIds() as $locale) {
```

Comment:
```suggestion
        foreach ($this->domain->getAllIds() as $domainId) {
```
and accordingly in usage below